README.md

```diff
@@ -26,93 +24,129 @@ API to generate signed URLs. See the [Signed URL](#url-signing) section for an e
 
-### Schemes
 
-The available authentication schemes align with relevant clients included in the [stac-asset](https://github.com/stac-utils/stac-asset) library.
+### Secure Asset Flow Object
+
+[OpenAPI OAuth Flow Object](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#oauth-flows-object). Allows configuration of the supported OAuth Flows.
+
+Configuration details for a supported OAuth Flow
+
+##### Fixed Fields
+Field Name | Type | Description
+---|:---:|---
+authorizationUrl | `string` | Required for `oauth2` (`"implicit"`, `"authorizationCode"`). The authorization URL to be used for this flow. This MUST be in the form of a URL.  |
+tokenUrl | `string` | Required for `oauth2` (`"password"`, `"clientCredentials"`, `"authorizationCode"`). The token URL to be used for this flow. This MUST be in the form of a URL.  |
+authorizationApi | `string` | Optional for `signedUrl`. The signed URL API endpoint to be used for this flow. If not enferred from the client environment, this must be defined in the authentication flow.  |
+refreshUrl | `string` | Optional for `oauth2`. The URL to be used for obtaining refresh tokens. This MUST be in the form of a URL.  |
+scopes | Map[`string`, `string`] | Required for `oauth2`. The available scopes for the authentication scheme. A map between the scope name and a short description for it. The map MAY be empty. |
 
-| Name                      | Description                                                                                                              |
-| ------------------------- | ------------------------------------------------------------------------------------------------------------------------ |
-| `HttpClient`              | Simple HTTP client without any authentication                                                                            |
-| `S3Client`                | Simple S3 client                                                                                                         |
-| `PlanetaryComputerClient` | Signs URLs with the [Planetary Computer Authentication API](https://planetarycomputer.microsoft.com/docs/reference/sas/) |
-| `EarthdataClient`         | Uses a token-based authentication to download data, from *some* Earthdata providers, e.g. DAACs                          |
-| `SignedUrlClient`         | Signs URLs with a user-defined authentication API                                                                      |
 
 ### URL Signing
 
-The `SignedUrlClient` scheme indicates that authentication will be handled by an API, which generates and returns a signed URL. For example, 
-a signed URL for an asset in AWS S3 can be generated with the following Lambda function code.
+The `signedUrl` scheme indicates that authentication will be handled by an API which generates and returns a signed URL. A signed URL authentication scheme can be defined with 
+```json
+{
+  "security:schemes": {
+    "signedUrl": {
+      "type": "signedUrl",
```

Comment:
This feels awkward, having the key be the same as the type. If we want to support multiple schemes with the same scheme (e.g. two signed URL options), then `security:schemes` should probably become a list?

Reply:
Agreed. The key can be any string. Perhaps `signedUrlAuth` or `signed_url_auth` are better.

Reply:
What's the point of the key again? Seems like if someone's trying to fetch a specific security scheme object, they're most likely to fetch by the scheme's name. Is there a reason to _not_ do a list?

Reply:
Ok yup the `security:refs` is the reason I forgot. Thanks for reminding me.